File: pset6/readability/readability.py
# Import Library
from cs50 import get_string
import re

# Ask for text input
text = get_string("Text: ")

# Obtengo el numero de oraciones
sent_num = max(len(re.split('! |\? |\. ', text)), 0)

# Get number of words in text
text_split = text.split()
num_words = len(text_split)

# Get all the letter
res = re.findall(r'\w+', text)
letras_cant = res
letras = 0

# Loop through each word
for word in res:
    cant = len(word)
    letras = letras + cant

# Calculate Coleman-Liau index
avg_let = ((letras) / num_words) * 100
avg_sent = ((sent_num) / num_words) * 100

level = round((0.0588 * avg_let) - (0.296 * avg_sent) - 15.8)

# Get Grade to Level
if level < 1:
    print('Before Grade 1')
elif level > 16:
    print('Grade 16+')
else:
    print('Grade', level)

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>
#include <math.h>

int main(void)
{
    //get input
    string text = get_string("Text: ");
    int num_letras = 0;
    int num_palabras = 1;
    int num_sent = 0;
    //loop to get letters
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (isalpha(text[i]))
        {
            //printf("%c",text[i]);
            num_letras ++;
        }
        if (text[i] == ' ' || text[i] == '\n' || text[i] == '\t')
        {
            num_palabras++;
    	}
    	if (text[i] == '!' || text[i] == '.' || text[i] == '?')
    	{
    	    num_sent ++;
    	}
    }
    //get average letter and sentences
    float avg_let = ((float)(num_letras) / (float)num_palabras) * 100;
    float avg_sent = ((float)(num_sent) / (float)num_palabras) * 100;

    //calculate
    int index = (int)round((0.0588 * avg_let) - (0.296 * avg_sent) - 15.8);
    //print
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}